pr_agent/tools/pr_description.py

```diff
@@ -139,34 +140,45 @@ def _prepare_pr_answer(self) -> Tuple[str, str, List[str], str]:
         - title: a string containing the PR title.
         - pr_body: a string containing the PR body in a markdown format.
         - pr_types: a list of strings containing the PR types.
-        - markdown_text: a string containing the AI prediction data in a markdown format.
+        - markdown_text: a string containing the AI prediction data in a markdown format. used for publishing a comment
         """
         # Load the AI prediction data into a dictionary
-        data = json.loads(self.prediction)
+        data = load_yaml(self.prediction.strip())
 
         # Initialization
-        markdown_text = pr_body = ""
         pr_types = []
 
         # Iterate over the dictionary items and append the key and value to 'markdown_text' in a markdown format
+        markdown_text = ""
         for key, value in data.items():
             markdown_text += f"## {key}\n\n"
             markdown_text += f"{value}\n\n"
 
         # If the 'PR Type' key is present in the dictionary, split its value by comma and assign it to 'pr_types'
         if 'PR Type' in data:
-            pr_types = data['PR Type'].split(',')
+            if type(data['PR Type']) == list:
+                pr_types = data['PR Type']
+            elif type(data['PR Type']) == str:
+                pr_types = data['PR Type'].split(',')
```

Comment:
**Suggestion:** In the `prepare_pr_description` function, you are checking the type of `data['PR Type']` twice. You can simplify this by using `isinstance` function which checks if an object is an instance of a class or of a subclass thereof.
```suggestion
        if 'PR Type' in data:
            if isinstance(data['PR Type'], list):
                pr_types = data['PR Type']
            elif isinstance(data['PR Type'], str):
                pr_types = data['PR Type'].split(',')
```